src/OrbitQt/orbitmainwindow.h

```diff
@@ -95,7 +90,7 @@ class OrbitMainWindow : public QMainWindow {
 
   void RestoreDefaultTabLayout();
 
-  std::optional<orbit_qt::TargetConfiguration> ClearTargetConfiguration();
+  orbit_qt::TargetConfiguration ClearTargetConfiguration();
```

Comment:
If we are returning TargetConfiguration, should this function be [[nodiscard]]?

Reply:
Good catch, thank you!

Reply:
Well actually. Forget my last comment. I deliberately chose this not to be to be `[[nodiscard]]`. The result of this function is only used when the target can be reused. In the case that an error with the connection occurs, the target is discarded and a new connection has to be established. 
You can see this in `main.cpp:132`
I hope this explanation makes sense

Reply:
You could consider still making it nodiscard and explicitly discarding it in main.cpp when not needed. By doing this you make clear to others that you thought this code path and that it's not an oversight.

Reply:
I like that idea, I will do that